### ISSUE TITLE
torch.compile produces wrong gradient for custom autograd.Function when the same tensor is passed twice

### ISSUE THREAD

**Issue by ALinrunrun** [pytorch/pytorch#183604]
### 🐛 Describe the bug

`torch.compile` with the Inductor backend produces an incorrect gradient for a custom `torch.autograd.Function` when the same tensor is passed as two inputs.

Eager execution correctly accumulates the gradients from both input positions. However, the compiled version only returns half of the expected gradient.

```
import torch

class F1(torch.autograd.Function):
    @staticmethod
    def forward(ctx, a, b):
        ctx.save_for_backward(a, b)
        return a * b + a + b

    @staticmethod
    def backward(ctx, go):
        a, b = ctx.saved_tensors
        return go * (b + 1), go * (a + 1)

def fn(x):
    return F1.apply(x, x).sum()

x = torch.tensor([0.7, -1.3, 2.1, 0.05, -0.5], requires_grad=True)

fn(x).backward()
ge = x.grad.clone()

x.grad = None

torch._dynamo.reset()
torch.compile(fn, backend="inductor")(x).backward()
gc = x.grad.clone()

print("eager  :", ge.tolist())
print("compile:", gc.tolist())
```

### Error logs

eager  : [3.4000000953674316, -0.5999999046325684, 6.199999809265137, 2.0999999046325684, 1.0]
compile: [1.7000000476837158, -0.2999999523162842, 3.0999999046325684, 1.0499999523162842, 0.5]

### Versions

PyTorch version: 2.11.0+cu130
Is debug build: False
CUDA used to build PyTorch: 13.0
ROCM used to build PyTorch: N/A

OS: Ubuntu 24.04.2 LTS (x86_64)
GCC version: (Ubuntu 13.3.0-6ubuntu2~24.04.1) 13.3.0
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.39

Python version: 3.10.20 (main, Mar 11 2026, 17:46:40) [GCC 14.3.0] (64-bit runtime)
Python platform: Linux-5.15.167.4-microsoft-standard-WSL2-x86_64-with-glibc2.39
Is CUDA available: False
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 3080 Laptop GPU
Nvidia driver version: 545.92
cuDNN version: Could not collect
Is XPU available: False
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True
Caching allocator config: N/A

Versions of relevant libraries:
[pip3] numpy==2.2.6
[pip3] onnx==1.21.0
[pip3] onnx2torch==1.5.15
[pip3] onnxruntime==1.23.2
[pip3] torch==2.11.0
[pip3] torchvision==0.26.0
[pip3] triton==3.6.0

cc @ezyang @gchanan @kadeng @msaroufim @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @amjames @jataylo @azahed98

**Comment by anijain2305** [comment]
duplicate of https://github.com/pytorch/pytorch/issues/181146 

closing
